### DISCUSSION TITLE
Using Maven default directory structure for new plug-ins?

### DISCUSSION THREAD

**Discussion by vogella** [eclipse-pde/eclipse.pde#82] [Ideas]
Not sure if that was discussed in the past but maybe PDE should allow to define plug-ins using the default directory structure Maven and Gradle are using, with src/main/java and src/test/java.

Given that we trying to move more and more to proven standards which are wildly used, this might be helpful to make plug-in development easier to newcomers.

**Comment by HannesWell** [comment]
> Not sure if that was discussed in the past but maybe PDE should allow to define plug-ins using the default directory structure Maven and Gradle are using, with src/main/java and src/test/java.

Using `src/main/java` and `src/test/java` as src folders should already be possible. One just has to specify the corresponding paths of the src-entries in the .classpath and adjust the paths of the src-entries in the build.property file (and likely only bin.include the main sources).

The only thing that is not working is to have dependencies that are only visible for tests.

**Comment by vogella** [comment]
Thanks @HannesWell this discussion was also intended to discuss if the PDE project wizard should use this new structure for newly created plug-ins.

**Comment by merks** [comment]
I expect that changing the default will kind of invalidate or confuse all existing tutorials and documentations around this long-established technology and long-used pattern.  Providing an additional explicit choice wouldn't harm, but there are likely 100 other things that would benefit the users more than this.
